### PR TITLE
Initial Functions and Classes for ProteinMPNN Model

### PR DESCRIPTION
**Pull request by shubham-bari**
## Description

Fix #(issue)

This PR adds the first PyTorch model components for ProteinMPNN in DeepChem, laying the groundwork for structure-conditioned protein sequence design based on Dauparas et al. The implementation follows the reference architecture’s feature-extraction stage and integrates with existing DeepChem ProteinMPNN support. The two classes introduced in this PR are as follows:

- PositionalEncodings: Encodes relative residue offsets and same-chain membership for kNN graph edges into learnable edge features used by the ProteinMPNN encoder.
- ProteinFeaturesLayer: Builds a k-nearest-neighbor graph from backbone coordinates and computes normalized edge embeddings from RBF-encoded distances across 25 backbone atom pairs (including virtual Cβ), combined with the positional encodings above.

Unit tests cover layer initialization, kNN neighbor selection, masking behavior, RBF encoding, training-time coordinate augmentation, batched forward passes, and end-to-end compatibility with featurizer output.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/ProteinMPNN.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+            ``(batch, num_nodes, k, num_embeddings)``.
+        """
+        d = torch.clip(offset + self.max_relative_feature, 0,
+                       2 * self.max_relative_feature) * mask + (1 - mask) * (
+                           2 * self.max_relative_feature + 1)
+        d_onehot = F.one_hot(d.long(), 2 * self.max_relative_feature + 2)
+        return self.linear(d_onehot.float())
+
+
+class ProteinFeaturesLayer(nn.Module):
```
Comment (Harindhar10) [inline deepchem/models/torch_models/ProteinMPNN.py:75]: Please add it to layers.rst and init.py

File: deepchem/models/torch_models/ProteinMPNN.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    Encodes relative sequence offsets and chain membership between residue
+    pairs into edge features used by the ProteinMPNN encoder.
+
+    Parameters
+    ----------
+    num_embeddings : int
+        Output dimension of the positional encoding.
+    max_relative_feature : int, optional (default 32)
+        Maximum absolute relative residue offset to encode.
+
```
Comment (Harindhar10) [inline deepchem/models/torch_models/ProteinMPNN.py:25]: Please add a usage example for the class and its methods

File: deepchem/models/torch_models/ProteinMPNN.py (hunk 1 above)
Comment (shubham-bari) [reply deepchem/models/torch_models/ProteinMPNN.py:75]: Added to the rst and init code

File: deepchem/models/torch_models/ProteinMPNN.py (hunk 2 above)
Comment (shubham-bari) [reply deepchem/models/torch_models/ProteinMPNN.py:25]: Added a usage example

**Comment by Harindhar10** [comment]
LGTM
